tests/phpunit/integration/Core/Util/Migration_1_121_0Test.php

```diff
@@ -0,0 +1,146 @@
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Util\Migration_1_121_0;
+use Google\Site_Kit\Modules\Analytics_4\Settings as Analytics_Settings;
+use Google\Site_Kit\Tests\Fake_Site_Connection_Trait;
+use Google\Site_Kit\Tests\TestCase;
+
+class Migration_1_121_0Test extends TestCase {
+
+	/**
+	 * @var Context
+	 */
+	protected $context;
+
+	/**
+	 * @var Options
+	 */
+	protected $options;
+
+	/**
+	 * @var Analytics_Settings
+	 */
+	protected $analytics_settings;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->context            = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options            = new Options( $this->context );
+		$this->analytics_settings = new Analytics_Settings( $this->options );
+
+		$this->analytics_settings->register();
+
+		$this->set_legacy_options();
+		$this->delete_db_version();
+	}
+
+	public function get_new_migration_instance() {
+		return new Migration_1_121_0(
+			$this->context,
+			$this->options
+		);
+	}
+
+	public function test_register() {
+		$migration = $this->get_new_migration_instance();
+		remove_all_actions( 'admin_init' );
+
+		$migration->register();
+
+		$this->assertTrue( has_action( 'admin_init' ) );
+	}
+
+	public function test_migrate() {
+		$migration = $this->get_new_migration_instance();
+		remove_all_actions( 'admin_init' );
+
+		$this->reset_analytics_4_options();
+
+		$migrated_keys = array(
+			'accountID',
+			'adsConversionID',
+			'trackingDisabled',
+		);
+
+		$migration->migrate();
+
+		$legacy_settings      = get_option( 'googlesitekit_analytics_settings' );
+		$analytics_4_settings = $this->analytics_settings->get();
+
+		$this->assertEquals(
+			$this->filter_settings( $analytics_4_settings, $migrated_keys ),
+			$this->filter_settings( $legacy_settings, $migrated_keys )
+		);
+
+		$this->assertEquals( Migration_1_121_0::DB_VERSION, $this->get_db_version() );
+	}
+
+	protected function filter_settings( $settings, $keys_to_filter ) {
+		return array_filter(
+			$settings,
+			function( $key ) use ( $keys_to_filter ) {
+				if ( in_array( $key, $keys_to_filter, true ) ) {
+					return true;
+				}
+
+				return false;
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+	}
+
+	protected function reset_analytics_4_options() {
+		// Set initial Analytics 4 module settings.
+		$this->analytics_settings->merge(
+			array(
+				'accountID'        => '', // Simulate an empty account id.
+				'propertyID'       => '987654321',
+				'webDataStreamID'  => '1234567890',
+				'measurementID'    => 'G-A1B2C3D4E5',
+				'trackingDisabled' => array( 'loggedinUsers' ),
+				'useSnippet'       => true,
+				'adSenseLinked'    => false,
+				'ownerID'          => get_current_user_id(),
+			)
+		);
+	}
+
+	protected function set_legacy_options() {
+		$this->options->set(
+			'googlesitekit_analytics_settings',
+			array(
+				'ownerID'               => get_current_user_id(),
+				'accountID'             => '12345678',
+				'adSenseLinked'         => true,
```

Comment:
The legacy `analytics` module still uses the `adsenseLinked` setting (not `adSenseLinked`).

```suggestion
				'adsenseLinked'         => true,
```